app/src/vanilla/java/cm/aptoide/pt/view/AppCoinsInfoManager.java

```diff
@@ -0,0 +1,23 @@
+package cm.aptoide.pt.view;
+
+import cm.aptoide.pt.app.view.AppCoinsInfoFragment;
+import cm.aptoide.pt.install.InstallManager;
+import rx.Observable;
+
+/**
+ * Created by D01 on 03/08/2018.
+ */
+
+public class AppCoinsInfoManager {
+
+  private final InstallManager installManager;
+
+  public AppCoinsInfoManager(InstallManager installManager) {
+
+    this.installManager = installManager;
+  }
+
+  public Observable<Boolean> loadButtonState() {
+    return installManager.isInstalled(AppCoinsInfoFragment.APPCWALLETPACKAGENAME);
+  }
+}
```

Comment:
I think that this class could be removed, because it doesn't add value. I understand that in the future if we put more logic in this screen we might need this "interactor aggregator" but today the only interactor that the view needs is the InstallManager, so the presenter should use that instead.